bql/semantic/expression.go

```diff
@@ -81,37 +81,44 @@ func (o OP) String() string {
 	}
 }
 
-// formatCell formats a given cell into a trimmed comparable string
-func formatCell(c *table.Cell) string {
+// formatCell formats a given cell into a trimmed comparable string.
+func formatCell(c *table.Cell) (string, error) {
 	if c.L != nil {
-		return strings.TrimSpace(c.L.ToComparableString())
+		return strings.TrimSpace(c.L.ToComparableString()), nil
 	}
-	return strings.TrimSpace(c.String())
+	if c.S != nil {
+		formatted, err := literal.DefaultBuilder().Build(literal.Text, *c.S)
+		if err != nil {
+			return "", fmt.Errorf("could not build a literal from string while trying to formatCell, got error: %v", err)
```

Comment:
You can add information on c.S so this error is more useful - i.e. help the maintainer to understand where the error happened too.

Reply:
Done. Thank you Tati.